app/controllers/ops_controller/settings/upload.rb

```diff
@@ -115,4 +115,27 @@ def logo_dir
     Dir.mkdir(dir) unless dir.exist?
     dir.to_s
   end
+
+  def valid_file?(file, type)
+    ext = file.original_filename.split(".").last.downcase  # Alternatively you can use File.extname
```

Comment:
haha this comment was for you not for the commit 😉 

The alternative would be something like `ext = File.extname(file.original_filename).downcase`, but it will include the `.` in it, so you need to check with `ext == ".#{type}`.

They were both kind of gross, but it's up to you.